### PR TITLE
Add hooks for adding a mo via crd

### PR DESCRIPTION
**Pull request by jojimt**
Define an interface and access functions to add delete a GBP mo, based on a CRD.

### CODE REVIEW

**Review by jayaramsatya — APPROVED** [review]
LGTM, Separate commit is required for dynamic detection CRD's similar to ACI controller in GBP server

File: pkg/gbpserver/crd_mo.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+package gbpserver
+
+import (
+	log "github.com/sirupsen/logrus"
+)
+
+// CRDMo acts as the interface between a CRD and the corresponding MO.
+// Implement this interface in the CRD watcher
+type CRDMo interface {
```
Comment (tomflynn) [inline pkg/gbpserver/crd_mo.go:25]: Can we rename to something like OpflexMo or GenieMo?

Comment (jojimt) [reply pkg/gbpserver/crd_mo.go:25]: That's too generic. All MOs are Opflex/Genie MOs. How about CustomMo or FeatureMo? @tomflynn

Comment (tomflynn) [reply pkg/gbpserver/crd_mo.go:25]: I figured this interface *could* be used for any opflex MO but I guess something like CustomOpflexMo is fine.

**Review by tomflynn — APPROVED** [review]

**Pull request merged**